### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by tezc**

### CODE REVIEW

File: include/raft.h [hunk 1]
```diff
@@ -926,7 +927,12 @@ void raft_clear(raft_server_t* me);
  *
  * On a restart, the application should set term and vote after reading metadata
  * file from the disk. See `raft_persist_metadata_f`.
- **/
+ *
+ * @param[in] raft The Raft server
+ * @param[in] term term in the metadata file
+ * @param[in] vote vote in the metadata file
+ * @return 0 on success
```
Comment (tezc) [inline include/raft.h:934]: I'm not a big fan of this comment style but following this one just to be consistent (as much as we can). Wdyt?

File: include/raft.h [hunk 2]
```diff
@@ -941,10 +947,6 @@ void raft_set_callbacks(raft_server_t* me, raft_cbs_t* funcs, void* user_data);
  *
  * If a voting node already exists the call will fail.
  *
- * @note The order this call is made is important.
- *  This call MUST be made in the same order as the other raft nodes.
- *  This is because the node ID is assigned depending on when this call is made
```
Comment (tezc) [inline include/raft.h:946]: This is not true anymore. Probably from years ago.

File: include/raft.h [hunk 3]
```diff
@@ -1682,7 +1711,7 @@ int raft_config(raft_server_t *me, int set, raft_config_e config, ...);
  *  cluster availability.
  *
  *  @param[in] raft The Raft server
```
Comment (fadidahanna) [inline include/raft.h:1713]:
```suggestion
 *  @param[in] me The Raft server
```

Comment (fadidahanna) [reply include/raft.h:1713]: Although I would prefer to change the parameter name (me->raft_server/server/etc)

File: include/raft.h [hunk 4]
```diff
@@ -1691,6 +1720,9 @@ int raft_pending_operations(raft_server_t *me);
  * This function should only be called after a restart. After application loads
  * the snapshot and log entries, this function should be called to rebuild
  * cluster configuration from the logs.
+ *
+ *  @param[in] raft The Raft server
+ *  @return 0 on success
```
Comment (fadidahanna) [inline include/raft.h:1725]: please add `RAFT_ERR_MISUSE` error case

File: include/raft.h [hunk 5]
```diff
@@ -1523,14 +1527,12 @@ void raft_entry_hold(raft_entry_t *ety);
  */
 void raft_entry_release(raft_entry_t *ety);
 
-/** Iterate an array of raft_entry_t* and release each element.
+/** Iterate an array of raft_entry_t* and release each element. Array itself
+ * will be freed at the end.
```
Comment (tezc) [inline include/raft.h:1531]: We free the array at the end. Probably, comment is from a previous version.

File: include/raft.h [hunk 6]
```diff
@@ -926,7 +927,12 @@ void raft_clear(raft_server_t* me);
...
  * On a restart, the application should set term and vote after reading metadata
  * file from the disk. See `raft_persist_metadata_f`.
- **/
+ *
+ * @param[in] raft The Raft server
+ * @param[in] term term in the metadata file
+ * @param[in] vote vote in the metadata file
+ * @return 0 on success
+ */
 int raft_restore_metadata(raft_server_t *me,
```
[the remark this answers is not in the document]
Comment (tezc) [reply include/raft.h:936]: Just to be future-proof. If we want to add a return code later, we can make it without breaking the API. Currently, there is no validation in it but maybe we should add, I am not sure right now.

File: include/raft.h (hunk 5 above)
Comment (fadidahanna) [reply include/raft.h:1531]: Yes, saw it in the code. Great you caught it in the documentation

File: include/raft.h [hunk 7]
```diff
@@ -1540,6 +1542,13 @@ extern const raft_log_impl_t raft_log_internal_impl;
 
 void raft_handle_append_cfg_change(raft_server_t* me, raft_entry_t* ety, raft_index_t idx);
 
+/** Enqueue a readonly request
+ *
+ * @param[in] me The Raft server
+ * @param[in] cb Function to be called when it is safe to execute the request
+ * @param[in] cb_arg User argument to the callback
+ * @return 0 on success
```
[the remark this answers is not in the document]
Comment (tezc) [reply include/raft.h:1394]: done

File: include/raft.h [hunk 8]
```diff
@@ -1548,17 +1557,37 @@ int raft_recv_read_request(raft_server_t* me, raft_read_request_callback_f cb, v
  *                    selection. Node with the most entries will be selected.
  * @param[in] timeout timeout in ms before this transfer is aborted.
  *                    if 0, use default election timeout
- * @return  an error if leadership transfer is already in progress or
- *          the targeted node_id is unknown
+ * @return 0 on success
```
[the remark this answers is not in the document]
Comment (tezc) [reply include/raft.h:1405]: done

File: include/raft.h (hunk 4 above)
Comment (tezc) [reply include/raft.h:1572]: done

**Review by fadidahanna — APPROVED** [review]

**Pull request merged**